src/page/personal-information.page.ts

```diff
@@ -0,0 +1,69 @@
+import { ElementFinder, element, by, browser, ExpectedConditions } from 'protractor';
+
+export class PersonalInformationPage {
+  private firstNameField: ElementFinder;
+  private lastNameField: ElementFinder;
+  private sexChkBox: ElementFinder;
+  private experienceChkBox: ElementFinder;
+  private professionChkbox: ElementFinder;
+  private toolsChkBox: ElementFinder;
+  private continentCmBox: ElementFinder;
+  private commandsCmBox: ElementFinder;
+  private submitBtn: ElementFinder;
+  private pageTitle: ElementFinder;
+  private timeout: number;
+  constructor () {
+    this.firstNameField = element(by.name('firstname'));
+    this.lastNameField = element(by.name('lastname'));
+    this.submitBtn = element(by.name('submit'));
+    this.pageTitle = element(by.cssContainingText('h1', 'Selenium - Automation Practice Form'));
+    this.timeout = 5000;
+  }
+
+  public async submit(userData) {
+    await this.fillForm(userData);
+    await this.submitBtn.click();
+
+    browser.wait(ExpectedConditions.alertIsPresent(), this.timeout);
+    await browser.switchTo().alert().accept();
+  }
+
+  public async getActualPageTitle(): Promise<string> {
+    await browser.wait(ExpectedConditions.visibilityOf(this.pageTitle), this.timeout);
+    return this.pageTitle.getText();
+  }
+
+  private async fillForm(userData): Promise<void> {
+
+    browser.wait(ExpectedConditions.visibilityOf(this.firstNameField), this.timeout);
+    await this.firstNameField.sendKeys(userData.firstName);
+    await this.lastNameField.sendKeys(userData.lastName);
+
+    this.sexChkBox = element(by.css(`input[name="sex"][value="${userData.sex}"]`));
```

Comment:
why not use the `$` shortcut?

Reply:
because at the beginning i had some errors with the locators, so i started to try with all the possibilities, and at the end, when i fixed the error, i forget to change to the shortcut.